superset/sql_lab.py

```diff
@@ -185,7 +186,8 @@ def handle_error(msg):
                 query.user_id, start_dttm.strftime('%Y_%m_%d_%H_%M_%S'))
         executed_sql = superset_query.as_create_table(query.tmp_table_name)
         query.select_as_cta_used = True
-    elif (query.limit and superset_query.is_select()):
+    elif (not query.limit and superset_query.is_select() and SQL_MAX_ROWS):
```

Comment:
Wait I think you remove all handling of `query.limit` where it's defined

Reply:
@mistercrunch  I tested it and it works correctly.

I changed the place where query.limit is defined upstream and it is None only when it is not specified. It is only in those cases that we append the default limit to the end.